### DISCUSSION TITLE
Help with using ki for rust development

### DISCUSSION THREAD

**Discussion by blitzerr** [ki-editor/ki-editor#322] [Q&A]
Can someone point me to configuration file format if they are using `ki` for their rust development.
Also is there a way to get all the key bindings ?

**Comment by wongjiahau** [comment]
Hey @blitzerr! There’s currently no configuration format available as it’s still in the planning. 

For keybindings unfortunately they are scattered all over the documentation, so I’m going to come up with something like a cheatsheet or recipes to combine most of them into a single page.

**Reply by blitzerr** [reply]
A cheat sheet would be perfect. I can also grep the ki source code if I know what to grep for. You all are doing incredible work with this editor. Keep going !

**Comment by wongjiahau** [comment]
Also, Ki already has built-in support for Rust, including:
1. Tree-sitter Rust
2. Auto format via cargo fmt
3. LSP using Rust Analzyer

**Reply by blitzerr** [reply]
So you are saying, based on the file type (in this case rust), ki will invoke rust-analyzer ? This is awesome. Maybe then I am doing something wrong. Or need to figure out how to invoke language server specific commands like auto-complete or say format on save. Can you point me to the code where this happens?

**Reply by wongjiahau** [reply]
Yup, it will start Rust analyzer once you open a Rust file, you do not need to invoke any command to start it.

Do you have `rust-analyzer` installed? Also `rust-analyzer` will take some time to index the repository, which might explain why you didn't get immediate feedback.

Can you share the log file after you open a Rust file in a Rust project?

You can use `ki log` to print the path of the log file.
